newIDE/app/src/ProjectsStorage/LocalProjectWriter.js

```diff
@@ -32,14 +34,23 @@ const writeJSONFile = (object: Object, filepath: string): Promise<void> => {
 };
 
 export default class LocalProjectWriter {
-  static saveProject = (project: gdProject): Promise<void> => {
-    const filepath = project.getProjectFile();
-    const projectPath = path.dirname(project.getProjectFile());
-    if (!filepath) {
-      return Promise.reject('Unimplemented "Save as" feature');
+  static saveProject = (
```

Comment:
Let's make another function called `saveProjectAs` (and wire it to MainFrame in LocalApp.js).

In this function, let's 
1) show the save dialog as you did, 
2) if no filepath is chosen (you get an empty string), this means that the user cancel. Return a promise that is resolved with false (to indicate that user cancelled): `Promise.resolve(false); return;`
3) otherwise, this means that the user chose a proper filename, so you can:
4) Launch the logic to save the project file (the code starting at `if (project.isFolderProject()) {`)

and finally, the missing part:
5) Use gd.ProjectResourcesCopier (example here: https://github.com/4ian/GDevelop/commit/6a014c8b880cd78503e9cbff88ad47e80008b287#diff-6b95b1fde995dec7ea9e58c1d6cf9d39R2231) to copy all the resources from the game to the new folder.